components/prism-cmake.js

```diff
@@ -0,0 +1,8 @@
RESSION|PDB_NAME|PDB_NAME_.*|PDB_OUTPUT_DIRECTORY|PDB_OUTPUT_DIRECTORY_.*|POSITION_INDEPENDENT_CODE|POST_INSTALL_SCRIPT|PREDEFINED_TARGETS_FOLDER|PREFIX|PRE_INSTALL_SCRIPT|PRIVATE_HEADER|PROCESSORS|PROCESSOR_AFFINITY|PROJECT_LABEL|PUBLIC_HEADER|REPORT_UNDEFINED_PROPERTIES|REQUIRED_FILES|RESOURCE|RESOURCE_LOCK|RULE_LAUNCH_COMPILE|RULE_LAUNCH_COMPILE|RULE_LAUNCH_COMPILE|RULE_LAUNCH_CUSTOM|RULE_LAUNCH_CUSTOM|RULE_LAUNCH_CUSTOM|RULE_LAUNCH_LINK|RULE_LAUNCH_LINK|RULE_LAUNCH_LINK|RULE_MESSAGES|RUNTIME_OUTPUT_DIRECTORY|RUNTIME_OUTPUT_DIRECTORY_.*|RUNTIME_OUTPUT_NAME|RUNTIME_OUTPUT_NAME_.*|RUN_SERIAL|SKIP_AUTOGEN|SKIP_AUTOMOC|SKIP_AUTORCC|SKIP_AUTOUIC|SKIP_BUILD_RPATH|SKIP_RETURN_CODE|SOURCES|SOURCE_DIR|SOURCE_DIR|SOVERSION|STATIC_LIBRARY_FLAGS|STATIC_LIBRARY_FLAGS_.*|STATIC_LIBRARY_OPTIONS|STRINGS|SUBDIRECTORIES|SUFFIX|SYMBOLIC|TARGET_ARCHIVES_MAY_BE_SHARED_LIBS|TARGET_MESSAGES|TARGET_SUPPORTS_SHARED_LIBS|TESTS|TEST_INCLUDE_FILE|TEST_INCLUDE_FILES|TIMEOUT|TIMEOUT_AFTER_MATCH|TYPE|TYPE|USE_FOLDERS|VALUE|VARIABLES|VERSION|VISIBILITY_INLINES_HIDDEN|VS_CONFIGURATION_TYPE|VS_COPY_TO_OUT_DIR|VS_CSHARP_.*|VS_DEBUGGER_COMMAND|VS_DEBUGGER_COMMAND_ARGUMENTS|VS_DEBUGGER_ENVIRONMENT|VS_DEBUGGER_WORKING_DIRECTORY|VS_DEPLOYMENT_CONTENT|VS_DEPLOYMENT_LOCATION|VS_DESKTOP_EXTENSIONS_VERSION|VS_DOTNET_REFERENCEPROP_.*|VS_DOTNET_REFERENCES|VS_DOTNET_REFERENCES_COPY_LOCAL|VS_DOTNET_REFERENCE_.*|VS_DOTNET_TARGET_FRAMEWORK_VERSION|VS_GLOBAL_.*|VS_GLOBAL_KEYWORD|VS_GLOBAL_PROJECT_TYPES|VS_GLOBAL_ROOTNAMESPACE|VS_GLOBAL_SECTION_POST_.*|VS_GLOBAL_SECTION_PRE_.*|VS_INCLUDE_IN_VSIX|VS_IOT_EXTENSIONS_VERSION|VS_IOT_STARTUP_TASK|VS_KEYWORD|VS_MOBILE_EXTENSIONS_VERSION|VS_RESOURCE_GENERATOR|VS_SCC_AUXPATH|VS_SCC_LOCALPATH|VS_SCC_PROJECTNAME|VS_SCC_PROVIDER|VS_SDK_REFERENCES|VS_SHADER_DISABLE_OPTIMIZATIONS|VS_SHADER_ENABLE_DEBUG|VS_SHADER_ENTRYPOINT|VS_SHADER_FLAGS|VS_SHADER_MODEL|VS_SHADER_OBJECT_FILE_NAME|VS_SHADER_OUTPUT_HEADER_FILE|VS_SHADER_TYPE|VS_SHADER_VARIABLE_NAME|VS_STARTUP_PROJECT|VS_TOOL_OVERRIDE|VS_USER_PROPS|VS_WINDOWS_TARGET_PLATFORM_MIN_VERSION|VS_WINRT_COMPONENT|VS_WINRT_EXTENSIONS|VS_WINRT_REFERENCES|VS_XAML_TYPE|WILL_FAIL|WIN32_EXECUTABLE|WINDOWS_EXPORT_ALL_SYMBOLS|WORKING_DIRECTORY|WRAP_EXCLUDE|XCODE_ATTRIBUTE_.*|XCODE_EMIT_EFFECTIVE_PLATFORM_NAME|XCODE_EXPLICIT_FILE_TYPE|XCODE_EXPLICIT_FILE_TYPE|XCODE_FILE_ATTRIBUTES|XCODE_LAST_KNOWN_FILE_TYPE|XCODE_PRODUCT_TYPE|XCODE_SCHEME_ADDRESS_SANITIZER|XCODE_SCHEME_ADDRESS_SANITIZER_USE_AFTER_RETURN|XCODE_SCHEME_ARGUMENTS|XCODE_SCHEME_DISABLE_MAIN_THREAD_CHECKER|XCODE_SCHEME_DYNAMIC_LIBRARY_LOADS|XCODE_SCHEME_DYNAMIC_LINKER_API_USAGE|XCODE_SCHEME_ENVIRONMENT|XCODE_SCHEME_EXECUTABLE|XCODE_SCHEME_GUARD_MALLOC|XCODE_SCHEME_MAIN_THREAD_CHECKER_STOP|XCODE_SCHEME_MALLOC_GUARD_EDGES|XCODE_SCHEME_MALLOC_SCRIBBLE|XCODE_SCHEME_MALLOC_STACK|XCODE_SCHEME_THREAD_SANITIZER|XCODE_SCHEME_THREAD_SANITIZER_STOP|XCODE_SCHEME_UNDEFINED_BEHAVIOUR_SANITIZER|XCODE_SCHEME_UNDEFINED_BEHAVIOUR_SANITIZER_STOP|XCODE_SCHEME_ZOMBIE_OBJECTS|XCTEST)\b/,
```

Comment:
Here, I'm guessing, but you probably don't want `.*` for two reasons:

1. I guess that `_OUTPUT_NAME` isn't a valid function, right? So please use the `+` operator. If it is, please ignore this point.
2. The dot `.` will match any character except line breaks. This includes spaces, brackets, punctuation, etc meaning that the `\b` you use to guard the properties doesn't mean much. In this case, you should probably use `\w` (== `[A-Za-z0-9_]`) instead of `.`.

Also, could you please group all words starting with `.*` (or `\w+`) together to get `.*(?:_OUTPUT_NAME|_POSTFIX|...)`?

All of these points also apply to `keyword`.

Reply:
Since the order is the same as generated from CMake's help `cmake --help-command-list` `cmake --help-property-list` `cmake --help-variable-list`, I propose to use the `\w+` approach and not group them.

Reply:
Hmmm, the CMake doc for `--help-command-list` says:

> [The list contains all commands for which help may be obtained by using the --help-command option followed by a command name.](https://cmake.org/cmake/help/v3.1/manual/ccmake.1.html#options)

Similar for the other ones. So are these lists complete?

Reply:
It covers much more than is used in everyday development. I will give it a spin on my blog and evaluate whether we need some additional commands/functions/keywords.

Reply:
Ok, so until proven otherwise, we'll just assume that they are complete. Works for me.